File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] map=new int[26];
        for(char c : s.toCharArray()){
            map[c-'a']++;
        }
        int pos=0;
        for(char c : s.toCharArray()){
            if(map[c-'a']==1) return pos;
            pos++;
        }
        return -1;
    }
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char,int> map;
        if(s.size()==1) return 0;
        for(auto c : s){
            map[c]++;
        }
        for(int i=0;i<s.size();i++){
            if(map[s[i]]==1) return i;
        }
        return -1;
    }
};